### ISSUE TITLE
EF forgets how to serialize an enum in a JsonColumn, you have to remind it and the error message is not helpful

### ISSUE THREAD

**Issue by AlexBaggett** [dotnet/efcore#33843]
## File a bug

When creating a JsonColumn via OwnsMany with an array of an owned class that has an enum property, EFCore will give an error message on attempts to run dotnet ef commands. This particular error message is wrong and unhelpful.

So basically this code causes problems because EF seems to forget how to serialize an enum to a string or int in the context of a JsonColumn.

### Include your code
```
public class SomeEntity
{
    public List<SomeOtherClass> Items = new ();
}
public  SomeOtherClass
{
   public SomeEnum SomeType {get; set;}
   public required string s {get; set;}
}
 public class SomeEntity_Map : IEntityTypeConfiguration<SomeEntity>
 {
        public void Configure(EntityTypeBuilder<SomeEntity> builder)
        {
            builder.OwnsMany(x => x.Items, x => x.ToJson());
        }
}
```

This is the error message EF generates.
```
Unable to create a 'DbContext' of type ''. The exception 'The entity type 'SomeOtherClass' cannot be configured as non-owned because it has already been configured as a owned. Use the nested builder in `OwnsOne` or `OwnsMany` on the owner entity type builder to further configure this type. If you want to override previous configuration first remove the entity type from the model by calling 'Ignore'. See https://aka.ms/efcore-docs-owned for more information and examples.' was thrown while attempting to create an instance. For the different patterns supported at design time, see https://go.microsoft.com/fwlink/?linkid=851728
```

This useless error message is resolved by changing the code to do this: 
```
    public class SomeEntity_Map : IEntityTypeConfiguration<SomeEntity>
    {
        public void Configure(EntityTypeBuilder<SomeEntity> builder)
        {
            builder.OwnsMany(x => x.Items, x =>
            {
                x.ToJson();
                x.Property(y => y.SomeType).HasConversion<string>();
            });
        }
    }
```

Basically the issue is that EF doesn't know how to serialize the enum now that is in the context of Json for some reason.  The error message generated does nothing to clue us in that the is the issue. 

Ideally, I'd like either a fix or at least a more useful error message.

### Include provider and version information

EF Core version: 8.03
Database provider:  Microsoft.EntityFrameworkCore.SqlServer
Target framework: (e.g. .NET 8.0)
Operating system: Windows 11
IDE: (e.g. Visual Studio 2022 17.9.6

**Comment by maumar** [comment]
@AlexBaggett I am unable to reproduce this on my end. I'm using the following code:

```cs
public class SomeEntity
{
    public int Id { get; set; }
    public List<SomeOtherClass> Items = new();
}
public class SomeOtherClass
{
    public SomeEnum SomeType { get; set; }
    public required string s { get; set; }
}


public enum SomeEnum
{
    One,
    Two
}

public class SomeEntity_Map : IEntityTypeConfiguration<SomeEntity>
{
    public void Configure(EntityTypeBuilder<SomeEntity> builder)
    {
        builder.OwnsMany(x => x.Items, x =>
        {
            x.ToJson();
        });
    }
}

public class MyContext : DbContext
{
    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        modelBuilder.ApplyConfigurationsFromAssembly(typeof(SomeEntity_Map).Assembly);
    }

    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=Repro;Trusted_Connection=True;MultipleActiveResultSets=true");
    }
}
```

and the following dotnet ef command:

```
dotnet ef migrations add Mig1
```

Can you try upgrading to the latest patch (8.0.6) and if that doesn't work, try using Package Manager Console commands (need to add Microsoft.EntityFrameworkCore.Tools package)

We did some changes around enums, now by default they map to int rather than string: https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-8.0/breaking-changes#enums-as-ints

If you worked with EF Core 7 before that might be the culprit.
